headers/headers.go

```diff
@@ -0,0 +1,75 @@
+// headers package provides header name constants and get/set functions for commonly used http headers in the
+// dp-api-clients-go repo. Package replaces go-ns lib and should be treated as the single source of truth
+package headers
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/ONSdigital/log.go/log"
+)
+
+var (
+	// CollectionIDKey is the name used for a collection ID http request header
+	CollectionIDKey = "Collection-Id"
+
+	// ErrHeaderNotFound returned if the requested header is not present in the provided request
+	ErrHeaderNotFound = errors.New("header not found")
+
+	errRequestNil = errors.New("error setting request header request was nil")
+)
+
+func getRequestHeader(req *http.Request, headerName string) (string, error) {
+	if req == nil {
+		return "", errRequestNil
+	}
+
+	headerValue := req.Header.Get(headerName)
+	if len(headerValue) == 0 {
+		return "", ErrHeaderNotFound
+	}
+
+	return headerValue, nil
+}
+
+func setRequestHeader(req *http.Request, headerName string, headerValue string) error {
```

Comment:
Just a suggestion not an issue.
Alternativly could have
```suggestion
func setRequestHeader(req *http.Request, headerName, headerValue string) error {
```

Reply:
Personally not a fan of that style